### PR TITLE
Draconian/legacy password strength validation

### PR DESCRIPTION
**Pull request by millerdev**
For ICDS security audit. Review commits individually (the first is just cleanup).

@czue cc @proteusvacuum

### CODE REVIEW

File: corehq/apps/domain/forms.py [hunk 1]
```diff
@@ -59,29 +61,24 @@
...
-from corehq.apps.reminders.models import CaseReminderHandler
-
-from corehq.apps.users.models import WebUser, CommCareUser, CouchUser
 from corehq.apps.groups.models import Group
 from corehq.apps.hqwebapp.crispy import TextField
 from corehq.apps.hqwebapp.tasks import send_mail_async, send_html_email_async
+from corehq.apps.reminders.models import CaseReminderHandler
+from corehq.apps.sms.phonenumbers_helper import parse_phone_number
+from corehq.apps.style import crispy as hqcrispy
+from corehq.apps.style.forms.widgets import BootstrapCheckboxInput, BootstrapDisabledInput
```
Comment (dimagimon) [inline corehq/apps/domain/forms.py]: F401 'BootstrapDisabledInput' imported but unused

File: corehq/apps/domain/forms.py [hunk 2]
```diff
@@ -964,14 +965,49 @@ def save(self, domain):
...
+        c = unichr(i)
+        if c.isupper():
+            uppers.append(c)
+    uppers.append(u']')
+    upper_group = u"".join(uppers)
+    return re.compile(upper_group, re.UNICODE)
+
+SPECIAL = re.compile(ur"\W", re.UNICODE)
+NUMBER = re.compile(ur"\d", re.UNICODE)  # are there other unicode numerals?
+UPPERCASE = _get_uppercase_unicode_regexp()
```
Comment (czue) [inline corehq/apps/domain/forms.py]: i didn't double check your regex logic but assume that since you've written a text editor that i didn't need to

**Comment by czue** [comment]
looks good to me and thanks for the tests. am i right that if this setting is enabled the front-end notification will not always be accurate? I think that's fine for now since we don't expect to turn this on ~anywhere except NIC but just wanted to check

File: corehq/apps/domain/forms.py (hunk 2 above)
Comment (millerdev) [inline corehq/apps/domain/forms.py]: haha, you should never assume. a regex can never have too many eyes.

**Comment by millerdev** [comment]
Can you elaborate on how the notification will not be accurate? I added a [different message](https://github.com/dimagi/commcare-hq/pull/10864/commits/6eae7d064a59723d772570e1eb55cb775a961c0f) for validation failure when this is enabled. Is that message not displayed on the front-end? I assumed it is since it is translated.

**Comment by czue** [comment]
I think there is also a javascript component to the validation on the front end and didn't see where that could have been updated. @calellowitz could say more or you could just go digging for it. like I said I don't think it's a blocker even if it is no longer accurate

**Comment by czue** [comment]
(though one simple thing would be maybe to just disable it if you have the setting enabled?)

**Comment by calellowitz** [comment]
the front end validation cory referenced is https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/registration/static/registration/js/password.js
I take it this means ICDS changed their mind about allowing us to use entropy based password strength

**Comment by czue** [comment]
@millerdev what do you think about disabling front-end validation notifications if this setting is enabled?

**Comment by millerdev** [comment]
Assume you're talking about the password strength indicator? Sure, I'll add it to my TODO list.

**Comment by millerdev** [comment]
@czue: 40004f3

**Comment by czue** [comment]
just waiting on tests. :boot:

**Pull request merged**